mayastor/src/bdev/dev/nvmx/channel.rs

```diff
@@ -63,24 +65,75 @@ pub struct NvmeIoChannelInner<'a> {
     pub qpair: *mut spdk_nvme_qpair,
     poll_group: NonNull<spdk_nvme_poll_group>,
     poller: poller::Poller<'a>,
+    // Flag to indicate the shutdown state of the channel.
+    // We need such a flag to differentiate between channel reset and shutdown.
+    // Channel reset is a reversible operation, which is followed by
+    // reinitialize(), which 'resurrects' channel (i.e. recreates all its
+    // I/O resources): such behaviour is observed during controller reset.
+    // Shutdown, in contrary, means 'one-way' ticket for the channel, which
+    // doesn't assume any further resurrections: such behaviour is seen
+    // upon controller shutdown. Being able to differentiate between these
+    // 2 states allows controller reset to behave properly in parallel with
+    // shutdown (if case reset is initiated before shutdown), and
+    // not to reinitialize channels already processed by shutdown logic.
+    is_shutdown: bool,
 }
 
 impl NvmeIoChannelInner<'_> {
-    /// Resets channel, making it unusable till reinitialize() is called.
+    /// Reset channel, making it unusable till reinitialize() is called.
     pub fn reset(&mut self) -> i32 {
         if self.qpair.is_null() {
             return 0;
         }
 
-        0
+        self._abort_queue();
+
+        let q = self.qpair;
```

Comment:
nit; we could use `std::ptr::swap()`  but likely its better that we abstract all of this in `qp.rs` if you will just as we did with the poller (i can work on this)